### PR TITLE
Admin site

### PR DESCRIPTION
**Pull request by thaidoandat**
## Related Tickets
- [#39837](https://edu-redmine.sun-asterisk.vn/issues/39837)

## WHAT (optional)
- Admin layout
- Log in, logout admin
- Search company basic
- Show list company

## Evidence (Screenshot or Video)
Screenshot from 2021-08-16 15-20-43
Screenshot from 2021-08-16 15-20-29
Screenshot from 2021-08-16 15-20-36

### CODE REVIEW

**Comment by hau1862** [comment]
ok

**Comment by letterforyou26** [comment]
ok

**Comment by thaidoandat** [comment]
ready

File: app/controllers/admin_companies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class AdminCompaniesController < AdminController
+  layout "admin"
```
Comment (ngocvt-0484) [inline app/controllers/admin_companies_controller.rb]: kế thừa từ Admin Controller rồi thì cần set layout k?
bỏ đi dc k

Comment (thaidoandat) [reply app/controllers/admin_companies_controller.rb]: bỏ được ạ

File: app/controllers/admin_sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  def destroy
+    log_out if logged_in?
+    redirect_to new_admin_session_path
+  end
+
+  private
+
+  def find_account_by_email
+    @account = Account.find_by email: params[:session][:email].downcase,
```
Comment (ngocvt-0484) [inline app/controllers/admin_sessions_controller.rb]: Account.admin.find_by email: params[:session][:email].downcase

File: app/models/company.rb [hunk 3]
```diff
@@ -22,4 +22,8 @@ class Company < ApplicationRecord
                      maximum: Settings.companies.description.length.max}
 
   delegate :email, :avatar, to: :account
+
+  def self.search key
+    key ? where("name LIKE ?", "%#{key}%") : all
```
Comment (ngocvt-0484) [inline app/models/company.rb]: viết scope by_name nhé e
k cần class method

File: app/models/job.rb [hunk 4]
```diff
@@ -60,4 +60,8 @@ def self.search_by search_params
           .search_by_category_ids(search_params[:categories])
           .includes(:company)
   end
+
+  def self.search key
+    key ? where("name LIKE ?", "%#{key}%") : all
```
Comment (ngocvt-0484) [inline app/models/job.rb]: same

File: app/views/layouts/_admin_topnav.html.erb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+          </a>
+        </li>
+      </ul>
+      <ul class="navbar-nav align-items-center  ml-auto ml-md-0 ">
+        <li class="nav-item dropdown">
+          <%= link_to admin_logout_path, method: :delete, class:"nav-link pr-0" do %>
+            <div class="media align-items-center">
+              <div class="media-body  ml-2  d-none d-lg-block">
+                <span class="mb-0 text-sm  font-weight-bold"><i class="fa fa-sign-out"></i></span>
+                <span class="mb-0 text-sm  font-weight-bold">Log out</span>
```
Comment (ngocvt-0484) [inline app/views/layouts/_admin_topnav.html.erb]: i18n

File: app/views/layouts/_admin_topnav.html.erb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <!-- Search form -->
+      <form class="navbar-search navbar-search-light form-inline mr-sm-3" id="navbar-search-main">
+        <div class="form-group mb-0">
+          <div class="input-group input-group-alternative input-group-merge">
+            <div class="input-group-prepend">
+              <span class="input-group-text"><i class="fa fa-search"></i></span>
+            </div>
+            <%= form_tag(admin_companies_path, method: :get) do %>
+              <%= text_field_tag :q, nil, placeholder: t(".search"), class: "form-control" %>
```
Comment (ngocvt-0484) [inline app/views/layouts/_admin_topnav.html.erb]: để tên là name dc k?
vì e search theo name mà

File: app/views/layouts/_admin_topnav.html.erb [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  <div class="container-fluid">
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <!-- Search form -->
+      <form class="navbar-search navbar-search-light form-inline mr-sm-3" id="navbar-search-main">
+        <div class="form-group mb-0">
+          <div class="input-group input-group-alternative input-group-merge">
+            <div class="input-group-prepend">
+              <span class="input-group-text"><i class="fa fa-search"></i></span>
+            </div>
+            <%= form_tag(admin_companies_path, method: :get) do %>
```
Comment (ngocvt-0484) [inline app/views/layouts/_admin_topnav.html.erb]:
```suggestion
            <%= form_tag admin_companies_path, method: :get  do %>
```

File: app/models/company.rb [hunk 8]
```diff
@@ -7,6 +7,8 @@ class Company < ApplicationRecord
 
   accepts_nested_attributes_for :account, update_only: true
 
+  scope :by_name, ->(name){where("name Like ?", "%#{name}%")}
```
Comment (ngocvt-0484) [inline app/models/company.rb:10]:
```suggestion
  scope :by_name, ->name{where("name Like ?", "%#{name}%") if name.present?}
```

File: app/models/job.rb [hunk 9]
```diff
@@ -41,7 +41,7 @@ class Job < ApplicationRecord
       .having("COUNT(job_categories.category_id) = ?", category_ids.count)
       .distinct
   end)
-
+  scope :by_name, ->(name){where("name Like ?", "%#{name}%")}
```
Comment (ngocvt-0484) [inline app/models/job.rb:44]: same

**Pull request merged**